docs/streams/developer-guide/dsl-topology-naming.html

```diff
@@ -0,0 +1,290 @@
+		 processor shifts, hence shifting the name of the processor.  Since <strong>most</strong> processors exist 
+		 in memory only this name shifting on topology changes presents no issue.  But the name 
+		 shifting does have implications for topologies with stateful operators or repartition topics. 
+
+		 Let's look at a different topology with some state:
+		 <code>
+		  builder.&gt;String,String&lt;stream("input").groupByKey().count().toStream().to("output");
+		 </code>
+
+		 This topology description yields the following:
+		 <code>
+		     Topologies:
+		   Sub-topology: 0
+		    Source: KSTREAM-SOURCE-0000000000 (topics: [input])
+		      --> KSTREAM-AGGREGATE-0000000002
+		    Processor: KSTREAM-AGGREGATE-0000000002 (stores: [KSTREAM-AGGREGATE-STATE-STORE-0000000001])
+		      --> KTABLE-TOSTREAM-0000000003
+		      <-- KSTREAM-SOURCE-0000000000
+		    Processor: KTABLE-TOSTREAM-0000000003 (stores: [])
+		      --> KSTREAM-SINK-0000000004
+		      <-- KSTREAM-AGGREGATE-0000000002
+		    Sink: KSTREAM-SINK-0000000004 (topic: output)
+		      <-- KTABLE-TOSTREAM-0000000003
+
+		  </code>
+		  
+		  We can see from the topology description above the state store is named 
+		  <code>KSTREAM-AGGREGATE-STATE-STORE-0000000002</code>.  But let's see what happens when we 
+		  add a filter to keep some of the records out of the aggregation:
+		  <code>
+		    builder.&gt;String, String&lt;stream("input")
+		        .filter((k,v)-> v !=null && v.length() >= 6 )
+		        .groupByKey()
+		        .count()
+		        .toStream()
+		        .to("output"); 
+		  </code>  
+
+		  And the corresponding topology:
+		  <code>
+		  Topologies:
+		   Sub-topology: 0
+		    Source: KSTREAM-SOURCE-0000000000 (topics: [input])
+		      --> KSTREAM-FILTER-0000000001
+		    Processor: KSTREAM-FILTER-0000000001 (stores: [])
+		      --> KSTREAM-AGGREGATE-0000000003
+		      <-- KSTREAM-SOURCE-0000000000
+		    Processor: KSTREAM-AGGREGATE-0000000003 (stores: [KSTREAM-AGGREGATE-STATE-STORE-0000000002])
+		      --> KTABLE-TOSTREAM-0000000004
+		      <-- KSTREAM-FILTER-0000000001
+		    Processor: KTABLE-TOSTREAM-0000000004 (stores: [])
+		      --> KSTREAM-SINK-0000000005
+		      <-- KSTREAM-AGGREGATE-0000000003
+		    Sink: KSTREAM-SINK-0000000005 (topic: output)
+		      <-- KTABLE-TOSTREAM-0000000004
+		  </code>
+
+		  Notice that since we've added an operation before the <code>count</code> operation, the state 
+		  store (hence the changelog topic as well) names have changed.  This name change means you can't 
+		  do a rolling re-deployment of your updated topology.  Additionally you'll have to use the  
+		  <a href="/{{version}}/documentation/streams/developer-guide/app-reset-tool">Streams Reset Tool</a> 
+		  to re-calculate the aggregations as since the changelog topic has changed on start-up the 
+		  new changelog topic contains no data.
+
+		  Fortunately, there is an easy solution to remedy this situation.  If we give a user-defined name 
```

Comment:
```suggestion
		  Fortunately, there's an easy solution to remedy this situation. Give a user-defined name 
```

Reply:
"Give the state store a constant user-defined name instead of relying..."